### PR TITLE
Prevent slot param value modification

### PR DESCRIPTION
**Pull request by wuelcas**
Hi,

I was using your framework and ran into a situation I could not understand at first and I think this PR solve it. Let's say we have this test

```javascript
describe('My Intent', () => {
  const mySlots = {
    place: 'Chicago',
  };

  describe('Some test', () => {
    alexaTest.test([
      {
        request: alexaTest.getIntentRequest('MyCustomIntent', mySlots),
      }
    ]);
  });

  describe('Some other test', () => {
    alexaTest.test([
      {
        request: alexaTest.getIntentRequest('MyCustomIntent', mySlots),
      }
    ]);
  });

  describe('One more', () => {
    alexaTest.test([
      {
        request: alexaTest.getIntentRequest('MyCustomIntent', mySlots),
      }
    ]);
  });
});
```

The tests will fail because the code in the `getIntentRequest` method is modifying the slot param, so also it modifies the value of the variable `mySlot` in the test above resulting in something like this:

```javascript
{
  name: 'place',
  value: {
    name: 'place',
    value: 'Chicago'
  }
}
```

This PR is to avoid this side effect

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -198,15 +198,17 @@ module.exports = {
 	/**
 	 * Generates an intent request object.
 	 * @param {string} intentName The name of the intent to call.
-	 * @param {object} slots Slot data to call the intent with.
+	 * @param {object} requestSlots Slot data to call the intent with.
```
Comment (hoegertn) [inline index.js]: Why did you rename the param?

File: index.js [hunk 2]
```diff
@@ -198,15 +198,17 @@ module.exports = {
...
-	getIntentRequest: function (intentName, slots, locale) {
+	getIntentRequest: function (intentName, requestSlots, locale) {
 		'use strict';
-		if (!slots) {
+		var slots;
+		if (!requestSlots) {
 			slots = {};
 		}
 		else {
+			slots = Object.create(requestSlots);
```
Comment (hoegertn) [inline index.js]: better use: `obj = JSON.parse(JSON.stringify(o));` and not prototype creation

File: index.js (hunk 1 above)
Comment (wuelcas) [reply index.js]: Well, I think I needed to rename the param name or the variable inside the method in order to make this work. I decided to rename the param to avoid changing the code too much.

File: index.js (hunk 2 above)
Comment (wuelcas) [reply index.js]: Good idea, I'll make the change right now

File: index.js (hunk 1 above)
Comment (hoegertn) [reply index.js]: Generally, I think renaming code inside a function is better as nobody relies on it in any way.

Comment (wuelcas) [reply index.js]: Well, I think either way works for this specific scenario. I'll change the name of the variable inside the function instead of the param name if that's better for you, no problem 👍
